pagopa/io-app
# chore: [IOAPPFD0-81] Update README.md

## Short description
This PR updates the README.md of the repository.

## List of changes proposed in this pull request
-  Removes `Theming`, `Custom UI component` and `Io-Icon-Font` because deprecated;
- Removes `Development with Backend App and Local Test IDP` and `Update the app icons` because outdated;
- Updates the reference to [PagoPA S.p.A.](https://www.pagopa.gov.it/) in `Who develops the app?`;
- Removes the email address from `How can I help you section" because it's invalid;
- Removes an old Pivotal thread from `iOS permission`;
- Puts these long permission tables under sections;
- Removes Native Base from `Main technologies used`;
- Moves the E2E README.md to the relevant folder; 
- Rewrites the build instructions in an easier way;
- Refactors the whole structure of the document;
- Adds icons/pictograms/logos README.md references in `Vector graphics`;
- Moves the `Internationalization` section to a new README.md placed in the relevant folder;
- Adds an `Appendix` section for grouping other README.md(s) placed in the relevant folders; 
- Removes the `Build (development)` with `fastlane` commands because you cannot run them anyway;
- Updates and rewords Android permissions;
- Rewords iOS premissions;
- Moves `Fonts` and `Vector graphics` to an appendix README.md;
- Replaces the outdated `Installation on physical devices (development)` section with the official documentation and a warning about changing the `Bundler Indetifier` to something unique;
- Adds supported deep linking URLs.

I tried to keep each commit as atomic as possible so you can track down each change by looking at the commits history.

## How to test
You can check the markdown preview [here](https://github.com/pagopa/io-app/blob/IOAPPFD0-81-update-readme/README.md).
